test/e2e/controller/etcd_test.go

```diff
@@ -0,0 +1,689 @@
+				logger.Info("successfully deleted Etcd")
+
+				logger.Info("finished running tests")
+
+				if !retainTestArtifacts {
+					cleanupTestArtifactsIfNecessary(testEnv, logger, g, testNamespace, etcd, timeoutEtcdReconciliation)
+				}
+			})
+		}
+	}
+}
+
+// TestScaleOut tests scale out of an Etcd cluster from 1 -> 3 replicas along with label changes.
+func TestScaleOut(t *testing.T) {
+	t.Parallel()
+	log := testr.NewWithOptions(t, testr.Options{LogTimestamp: true})
+
+	testCases := []struct {
+		name                         string
+		peerTLSEnabledBeforeScaleOut bool
+		labelsBeforeScaleOut         map[string]string
+		peerTLSEnabledAfterScaleOut  bool
+		labelsAfterScaleOut          map[string]string
+	}{
+		{
+			name: "basic",
+		},
+		{
+			name:                        "enable-peer-tls",
+			peerTLSEnabledAfterScaleOut: true,
+		},
+		{
+			name:                         "with-peer-tls",
+			peerTLSEnabledBeforeScaleOut: true,
+			peerTLSEnabledAfterScaleOut:  true,
+		},
+		{
+			name:                "with-label-change",
+			labelsAfterScaleOut: map[string]string{"foo": "bar"},
+		},
+		{
+			name:                        "enable-ptls-label-change",
+			peerTLSEnabledAfterScaleOut: true,
+			labelsAfterScaleOut:         map[string]string{"foo": "bar"},
+		},
+		// TODO: enable this once disabling peer TLS is supported
+		//{
+		//	name:                         "disable-peer-tls",
+		//	peerTLSEnabledBeforeScaleOut: true,
+		//},
+	}
+
+	g := NewWithT(t)
+
+	for _, provider := range providers {
+		for _, tc := range testCases {
+			tcName := fmt.Sprintf("scaleout-%s-%s", tc.name, provider)
+			t.Run(tcName, func(t *testing.T) {
+				t.Parallel()
+
+				testNamespace := testutils.GenerateTestNamespaceName(t, fmt.Sprintf("%s%s", testNamespacePrefix, tcName), 4)
+				logger := log.WithName(tcName).WithValues("etcdName", defaultEtcdName, "namespace", testNamespace)
+
+				initializeTestCase(g, testEnv, logger, testNamespace, defaultEtcdName)
+
+				logger.Info("running tests")
+				etcdBuilder := testutils.EtcdBuilderWithoutDefaults(defaultEtcdName, testNamespace).
+					WithReplicas(1).
+					WithClientTLS().
+					WithDefaultBackup().
+					WithBackupRestoreTLS().
+					WithStorageProvider(provider, fmt.Sprintf("%s/%s", testNamespace, defaultEtcdName))
+
+				if tc.peerTLSEnabledBeforeScaleOut {
+					etcdBuilder = etcdBuilder.WithPeerTLS()
+				}
+				if tc.labelsBeforeScaleOut != nil {
+					etcdBuilder = etcdBuilder.WithSpecLabels(tc.labelsBeforeScaleOut)
+				}
+				etcd := etcdBuilder.Build()
+
+				logger.Info("creating Etcd")
+				testEnv.CreateAndCheckEtcd(g, etcd, timeoutEtcdReconciliation)
+				logger.Info("successfully created Etcd")
+
+				logger.Info("scaling out Etcd to 3 replicas")
+				etcd.Spec.Replicas = 3
+				updateEtcdTLSAndLabels(etcd, false, tc.peerTLSEnabledAfterScaleOut, false, etcd.Spec.Labels)
+				testEnv.UpdateAndCheckEtcd(g, etcd, timeoutEtcdReconciliation*2)
```

Comment:
```suggestion
				testEnv.UpdateAndCheckEtcd(g, etcd, timeoutEtcdReconciliation)
```
Unless I'm missing something.